config/unicorn.conf.rb

```diff
@@ -154,72 +181,6 @@ def murder_lazy_workers
         next_sleep <= 0 ? 1 : next_sleep
       end
 
-      def max_sidekiq_rss
-        rss =
-          `ps -eo rss,args | grep sidekiq | grep -v grep | awk '{print $1}'`.split("\n")
-            .map(&:to_i)
-            .max
-
-        rss ||= 0
-
-        rss * 1024
-      end
-
-      def max_allowed_sidekiq_rss
-        [ENV["UNICORN_SIDEKIQ_MAX_RSS"].to_i, 500].max.megabytes
-      end
-
-      def force_kill_rogue_sidekiq
-        info = `ps -eo pid,rss,args | grep sidekiq | grep -v grep | awk '{print $1,$2}'`
-        info
-          .split("\n")
-          .each do |row|
-            pid, mem = row.split(" ").map(&:to_i)
-            if pid > 0 && (mem * 1024) > max_allowed_sidekiq_rss
-              Rails.logger.warn "Detected rogue Sidekiq pid #{pid} mem #{mem * 1024}, killing"
-              begin
-                Process.kill("KILL", pid)
-              rescue StandardError
-                nil
-              end
-            end
-          end
-      end
-
-      def check_sidekiq_heartbeat
-        @sidekiq_heartbeat_interval ||= 30.minutes
-        @sidekiq_next_heartbeat_check ||= Time.now.to_i + @sidekiq_heartbeat_interval
-
-        if @sidekiq_next_heartbeat_check < Time.now.to_i
-          last_heartbeat = Jobs::RunHeartbeat.last_heartbeat
-          restart = false
-
-          sidekiq_rss = max_sidekiq_rss
-          if sidekiq_rss > max_allowed_sidekiq_rss
-            Rails.logger.warn(
-              "Sidekiq is consuming too much memory (using: %0.2fM) for '%s', restarting" %
-                [(sidekiq_rss.to_f / 1.megabyte), ENV["DISCOURSE_HOSTNAME"]],
-            )
-
-            restart = true
-          end
-
-          if last_heartbeat < Time.now.to_i - @sidekiq_heartbeat_interval
-            Rails.logger.warn "Sidekiq heartbeat test failed, restarting"
-
-            restart = true
-          end
-          @sidekiq_next_heartbeat_check = Time.now.to_i + @sidekiq_heartbeat_interval
-
-          if restart
-            Demon::Sidekiq.restart
-            sleep 10
-            force_kill_rogue_sidekiq
-          end
-          Discourse.redis.close
-        end
-      end
```

Comment:
Moved most of these logic into `demon/sidekiq.rb` as these methods are not a concern of `Unicorn::HttpServer`.